tests/test_transformers.py

```diff
@@ -385,65 +387,82 @@ def test_resuming(self):
             warmup_updates=1,
         )
 
-        with testing_utils.tempdir() as tmpdir:
-            model_file = os.path.join(tmpdir, 'model')
+        RANKER_ARGS = dict(
+            task='integration_tests:candidate',
+            model='transformer/ranker',
+            optimizer='adamax',
+            learningrate=1e-3,
+            batchsize=32,
+            num_epochs=1,
+            n_layers=1,
+            n_heads=1,
+            ffn_size=32,
+            embedding_size=32,
+            warmup_updates=1,
+        )
 
-            stdout1, valid1, test1 = testing_utils.train_model(dict(
-                model_file=model_file,
-                lr_scheduler='invsqrt',
-                **BASE_ARGS,
-            ))
-            stdout2, valid2, test2 = testing_utils.train_model(dict(
-                model_file=model_file,
-                lr_scheduler='invsqrt',
-                **BASE_ARGS,
-            ))
-            # make sure the number of updates is being tracked correctly
-            self.assertGreater(
-                valid2['num_updates'],
-                valid1['num_updates'],
-                'Number of updates is not increasing'
-            )
-            # make sure the learning rate is decreasing
-            self.assertLess(
-                valid2['lr'],
-                valid1['lr'],
-                'Learning rate is not decreasing'
-            )
-            # but make sure we're not loading the scheduler if we're fine tuning
-            stdout3, valid3, test3 = testing_utils.train_model(dict(
-                init_model=os.path.join(tmpdir, 'model'),
-                model_file=os.path.join(tmpdir, 'newmodel'),
-                lr_scheduler='invsqrt',
-                **BASE_ARGS,
-            ))
-            self.assertEqual(
-                valid3['num_updates'],
-                valid1['num_updates'],
-                'Finetuning LR scheduler reset failed (num_updates).'
-            )
-            self.assertEqual(
-                valid3['lr'],
-                valid1['lr'],
-                'Finetuning LR scheduler reset failed (lr).'
-            )
-            # and make sure we're not loading the scheduler if it changes
-            stdout4, valid4, test4 = testing_utils.train_model(dict(
-                init_model=os.path.join(tmpdir, 'model'),
-                model_file=os.path.join(tmpdir, 'newmodel2'),
-                lr_scheduler='reduceonplateau',
-                **BASE_ARGS
-            ))
-            self.assertEqual(
-                valid4['num_updates'],
-                valid1['num_updates'],
-                'LR scheduler change reset failed (num_updates).\n' + stdout4
-            )
-            self.assertEqual(
-                valid4['lr'],
-                1e-3,
-                'LR is not correct in final resume.\n' + stdout4
-            )
+        for BASE_ARGS in [GENERATOR_ARGS, RANKER_ARGS]:
```

Comment:
As is, when this fails, it will be slightly annoying to tell whether it's the ranker model or the generator model that is failing.

It would be better to refactor out this code into a helper method, and then call it from two differently named tests with the proper args.

Reply:
if you look at my latest commit i print in the error message which model it is

Reply:
Alright I still think it be cleaner, but that’s what’s important :)